src/stream-ciphers.rst

```diff
@@ -137,43 +137,42 @@ significantly fewer pixels per block than the larger block sizes in the
-trivial.
+Notice that an idealized
+encryption scheme looks like random noise.
+“Looking like random noise” does not mean something is properly
+encrypted: it just means that we cannot inspect it using trivial methods.
 
 Encryption oracle attack
 ^^^^^^^^^^^^^^^^^^^^^^^^
 
-In the previous section, we've focused on how an attacker can inspect a
-ciphertext encrypted using :term:`ECB mode`. That's a *passive*, ciphertext-only
-attack. It's passive because the attacker doesn't really interfere in
-any communication; they're simply examining a ciphertext. In this
-section, we'll study a different, *active* attack, where the attacker
-actively communicates with their target. We'll see how the active attack
-can enable an attacker to decrypt ciphertexts encrypted using :term:`ECB mode`.
-
-To do this, we'll introduce a new concept called an :term:`oracle`. Formally
-defined :term:`oracle`\s are used in the study of computer science, but for our
-purposes it's sufficient to just say that an :term:`oracle` is something that
-will compute some particular function for you.
-
-In our case, the :term:`oracle` will perform a specific encryption for the
-attacker, which is why it's called an :term:`encryption oracle`. Given some data
-:math:`A` chosen by the attacker, the :term:`oracle` will encrypt that data,
+In the previous section, we focused on how an attacker can inspect a
+ciphertext encrypted using :term:`ECB mode`. That is a *passive*, ciphertext-only
+attack. It is passive because the attacker does not interfere in
+communications. The attacker simply examines the ciphertext. In this
+section, we study a different, *active* attack, where the attacker
+actively communicates with their target. We see how the active attack
+enables an attacker to decrypt ciphertexts encrypted using :term:`ECB mode`.
+
+To do this, we introduce a new concept called an :term:`oracle`. Formally
+defined :term:`oracle`\s are used in the field of computer science. For our
+purposes it is sufficient to say that an :term:`oracle`
+computes a particular function.
+
+In our case, the :term:`oracle` performs a specific encryption for an
+attacker as an :term:`encryption oracle`. Given some data
+:math:`A` chosen by the attacker, the :term:`oracle` encrypts that data,
 followed by a secret suffix :math:`S`, in :term:`ECB mode`. Or, in symbols:
 
 .. math::
 
    C = ECB(E_k, A \| S)
 
 The secret suffix :math:`S` is specific to this system. The attacker's
-goal is to decrypt it. We'll see that being able to encrypt other
-messages surprisingly allows the attacker to decrypt the suffix. This
-:term:`oracle` might seem artificial, but is quite common in practice. A simple
-example would be a cookie encrypted with ECB, where the prefix :math:`A`
-is a name or an e-mail address field, controlled by the attacker.
+goal is to decrypt it. Ability to encrypt other
+messages surprisingly allows the attacker to decrypt the suffix. The
```

Comment:
This should really be `this`: there are lots of kinds of oracles, this _specific_ one may seem artificial

Reply:
Got it. That makes sense